crates/biome_js_analyze/src/semantic_analyzers/correctness/use_exhaustive_dependencies.rs

```diff
@@ -524,6 +524,55 @@ fn is_out_of_function_scope(
     )
 }
 
+fn into_member_vec(node: &JsSyntaxNode) -> Vec<String> {
+    let mut vec = vec![];
+    let mut next = Some(node.clone());
+
+    while let Some(node) = &next {
+        match AnyJsMemberExpression::cast_ref(node) {
+            Some(member_expr) => {
+                let member_name = member_expr
+                    .member_name()
+                    .and_then(|it| it.as_string_constant().map(|it| it.to_owned()));
+                match member_name {
+                    Some(name) => {
+                        vec.insert(0, name);
+                        next = member_expr.object().ok().map(AstNode::into_syntax);
+                    }
+                    None => break,
+                }
+            }
+            None => {
+                vec.insert(0, node.text_trimmed().to_string());
+                break;
+            }
+        }
+    }
+
+    vec
+}
+
+fn compare_member_depth(a: &JsSyntaxNode, b: &JsSyntaxNode) -> (bool, bool) {
+    let mut a_member_iter = into_member_vec(a).into_iter();
+    let mut b_member_iter = into_member_vec(b).into_iter();
```

Comment:
Here's a suggestion. Instead of retuning a `Vec<String>`, why don't we create an iterator? Considering that we use `a_member_iter` and `b_member_iter` in a loop, we can achieve the same result without **allocating strings**.

Here's some meta code as a suggestion:

```rs
struct MemberIterator<'a> {
	current_node: &'a JsSyntaxNode,
	// a boolean that tells us when we have stop iterating
	has_members: true
}

impl Iterator for MemberIterator {
	type Item = TokenText

	fn next(&mut self) -> Option<Self::Item> {
		// here, we put more or less the same logic of into_member_vec
	}
}

impl FusedIterator  for MemberIterator {}
```